### ISSUE TITLE
Wagtail 2.15: Image renditions for re-oriented images are scaled and cropped using un-oriented dimensions

### ISSUE THREAD

**Issue by jmichalicek** [wagtail/wagtail#7700]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Renditions of images which need rotated by `willow.auto_orient()` are being created based on the dimensions of the unoriented image. For example an image from Android phone held in portrait creates the image with exif data stating the landscape dimensions and an orientation value of 6. So my phone says portrait photos are 4032x3024 with an orientation of 6.

This results in photos which are portraits being cut off at the bottom and having a large black bar on the right side.

I have done some digging and fixed this issue locally and so would like to make a PR with the fix. I have detailed what seems to be happening/need done to reesolve below.

### Steps to Reproduce

1. Take a picture with a device which can do portrait or landscape, but always saves the exif data for only one of those with an exif orientation value applied. I used a Samsung Galaxy S8 but I expect most phones are doing this.
2. Upload the image to wagtail images
3. Look at the images on the images list, in the image detail, and any generated for actual pages have the wrong dimensions

What appears to be the case is that the `willow.auto_orient()` call performs as expected and rotates the image. In the 2.15 code an ImageTransform is created and is used to get the rectangle dimensions of the image. Rather than using the re-oriented image, it is still working off of the original. The result is that it is cropping and scaling based on a 4032x3024 landscape photo in my case, so photos which are actually portrait get cropped based on those dimensions rather than the rotated dimensions of 3024x4032.

I have found that in `wagtail.images.models.Transform.run()` if I use the returned value from `willow.get_pillow_image()` after `willow.auto_orient()` is called everywhere which `image` which was passed into the method is currently used then everything appears to work correctly.

It does not look like that is going to work if Wand is being used since `get_pillow_image()` is specific to willow's `PillowImage` class. I will test just using `willow.image` this evening. My other thought would be to update Willow to add a plain `get_image()` to the plugins and register that with `@Image.operation` so that there is a consistently named operation to get at the image if not directly accessing `willow.image` is preferred.

I am attaching the actual original image and the rendition using the `original` format spec.

Original:
![wagtail_renditions_bug](https://user-images.githubusercontent.com/1059070/141117543-0d09245e-23b8-4052-8a45-b5d17138285e.jpg)

Rendition:
![wagtail_renditions_bug original](https://user-images.githubusercontent.com/1059070/141117572-06547567-e122-40ab-bf09-18499fba51a6.jpg)

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.10.0
* Django version: 3.2.9
* Wagtail version: 2.15
* Browser version: You can use https://www.whatsmybrowser.org/ to find this out.

**Comment by zerolab** [comment]
Reference note: transforms added in #6843

**Comment by jmichalicek** [comment]
Just passing in `willow.image` at https://github.com/wagtail/wagtail/blob/main/wagtail/images/models.py#L475 didn't quite work. While it solved my immediate issue, other filters and operations were relying on that being an actual `wagtail.images.models.Image` model, not just any object with `width` and `height` properties.

I have created a draft pull request with a proposed fix where I have added height and width arguments to `Filter.get_transform()`. I have also suggested some other additions/changes which may be desired on that pull request.
